.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -1191,6 +1390,291 @@ stages:
+      eq(dependencies.generate_variables.outputs['generate_variables_job.generate_variables_step.IsProfilerChanged'], 'True')
+    )
+  dependsOn: [master_commit_id, generate_variables, build_linux_tracer, build_linux_profiler]
+  variables:
+    masterCommitId: $[ stageDependencies.master_commit_id.fetch.outputs['set_sha.master']]
+  jobs:
+  - template: steps/update-github-status-jobs.yml
+    parameters:
+      jobs: [Test]
+
+  - job: Test
+    timeoutInMinutes: 60 #default value
+    strategy:
+      matrix:
+        debian:
+          baseImage: debian
+        alpine:
+          baseImage: alpine
+    
+        # Enable the Datadog Agent service for this job
+    services:
+      dd_agent: dd_agent
+
+    pool:
+      vmImage: ubuntu-18.04
+
+    steps:
+    - template: steps/install-dotnet-sdks.yml
+
+    - template: steps/clone-repo.yml
+      parameters:
+        masterCommitId: $(masterCommitId)
+
+    - template: steps/restore-working-directory.yml
+      parameters:
+        artifact: build-linux-$(baseImage)-working-directory
+  
+    - task: DownloadPipelineArtifact@2
+      inputs:
+        artifact: linux-profiler-binaries-$(baseImage)
+        path: $(System.DefaultWorkingDirectory)/profiler/_build
+      displayName: Download linux profiler binaries
+
+    - script: |
+        sudo apt-get update
+        sudo apt-get install -y libasan6
+      displayName: install libasan
+    
+    - script: |
+        CXX=clang++ CC=clang cmake -S profiler -B __build -DRUN_ASAN=ON
+        cmake --build __build --config Release --parallel
+      displayName: Build Native Profiler Engine
+
+    - script: |
+        dotnet build -c Release profiler/src/Demos/Datadog.Demos.sln -p:Platform=x64
+      displayName: build sample applications
+
+    - script: |
+        # set profiler deployment and test output dir folder path for the test
+        test_output_dir="$(realpath .)/test_output"
+
+        export DD_TESTING_OUPUT_DIR=$test_output_dir
+        export DD_PROFILING_LOG_DIR="$test_output_dir/logs"
+        export DD_PROFILING_OUTPUT_DIR="$test_output_dir/pprofs"
+        export DD_TRACE_DEBUG="1"
+
+        # Run Computer01 for 2min
+        source profiler/src/ProfilerEngine/DeployResources/DDProf-SetEnv.sh `pwd`/profiler/_build/DDProf-Deploy &&
+        cd profiler/_build/bin/Release-x64/profiler/src/Demos/Samples.Computer01/net5.0 &&
+        # LD_PRELOAD was previously set in the DDProf-SetEnv.sh script, but ASAN does not work with other
+        # library in LD_PRELOAD (segfault)
+        ASAN_OPTIONS="detect_leaks=0" LD_PRELOAD="libasan.so.6" dotnet Samples.Computer01.dll --scenario 1 --timeout 120
+      displayName: run asan test
+
+    - publish: $(System.DefaultWorkingDirectory)/test_output
+      displayName: Uploading test results
+      artifact: asan_tests_linux_profiler_$(baseImage)_$(System.JobAttempt)
+      condition: succeededOrFailed()
+      continueOnError: false
+
+- stage: ubsan_tests_linux_profiler
```

Comment:
What does `ubsan` stands for?

Reply:
If I got it right, it's some kind of code analysis.
I think @gleocadie can give more context on this.

Reply:
As said above, UBSAN == Undefined Behavior Sanitizer